auth/couchdb_session_authenticator_test.go

```diff
@@ -278,6 +284,75 @@ var _ = Describe("Authenticator Unit Tests", func() {
 
 		err = auth.Authenticate(request)
 		Expect(err).To(HaveOccurred())
-		Expect(string(authError)).To(Equal(err.Error()))
+		Expect(err.Error()).To(Equal(string(authError)))
+	})
+
+	It("Test missing AuthSession cookie in the response", func() {
+		server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			w.WriteHeader(http.StatusOK)
+		}))
+		defer server.Close()
+
+		builder, err := core.NewRequestBuilder(core.GET).
+			ResolveRequestURL(server.URL, "/db", nil)
+		Expect(err).To(BeNil())
+
+		request, err := builder.Build()
+		Expect(err).To(BeNil())
+		Expect(request).ToNot(BeNil())
+
+		auth, err := NewCouchDbSessionAuthenticator("user", "pass")
+		Expect(err).To(BeNil())
+		Expect(auth).ToNot(BeNil())
+		Expect(auth.AuthenticationType()).To(Equal(AUTHTYPE_COUCHDB_SESSION))
+
+		err = auth.Authenticate(request)
+		Expect(err).To(HaveOccurred())
+		Expect(err.Error()).To(HavePrefix("Missing AuthSession cookie"))
+	})
+
+	It("Test invalid format for AuthSession", func() {
+		server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			cookie := &http.Cookie{
+				Name: "AuthSession",
+				// "fake:fake" in base64
+				Value: "ZmFrZTpmYWtlCg==",
+			}
+			http.SetCookie(w, cookie)
+			w.WriteHeader(http.StatusOK)
+		}))
+		defer server.Close()
+
+		builder, err := core.NewRequestBuilder(core.GET).
+			ResolveRequestURL(server.URL, "/db", nil)
+		Expect(err).To(BeNil())
+
+		request, err := builder.Build()
+		Expect(err).To(BeNil())
+		Expect(request).ToNot(BeNil())
+
+		auth, err := NewCouchDbSessionAuthenticator("user", "pass")
+		Expect(err).To(BeNil())
+		Expect(auth).ToNot(BeNil())
+		Expect(auth.AuthenticationType()).To(Equal(AUTHTYPE_COUCHDB_SESSION))
+
+		err = auth.Authenticate(request)
+		Expect(err).To(HaveOccurred())
+		Expect(err.Error()).To(HavePrefix("Invalid format for AuthSession"))
+	})
+
+	It("Test requestSession failures", func() {
```

Comment:
This test could be split up, one for `test service url empty` and another for `connection refused`.

Reply:
Good idea, done in https://github.com/IBM/cloudant-go-sdk/pull/74/commits/4227d99225b23e9fa7cc6af6a4667c5c89d85a4f